crates/router/src/compatibility/stripe/app.rs

```diff
@@ -9,12 +9,24 @@ impl PaymentIntents {
     pub fn server(state: routes::AppState) -> Scope {
         web::scope("/payment_intents")
             .app_data(web::Data::new(state))
-            .service(payment_intents_retrieve_with_gateway_creds)
-            .service(payment_intents_create)
-            .service(payment_intents_retrieve)
-            .service(payment_intents_update)
-            .service(payment_intents_confirm)
-            .service(payment_intents_capture)
+            .service(
+                web::resource("/sync")
+                    .route(web::post().to(payment_intents_retrieve_with_gateway_creds)),
+            )
+            .service(web::resource("").route(web::post().to(payment_intents_create)))
+            .service(
+                web::resource("/{payment_id}")
+                    .route(web::get().to(payment_intents_retrieve))
+                    .route(web::post().to(payment_intents_update)),
+            )
+            .service(
+                web::resource("/{payment_id}/confirm")
+                    .route(web::post().to(payment_intents_confirm)),
+            )
+            .service(
+                web::resource("/{payment_id}/capture")
+                    .route(web::post().to(payment_intents_capture)),
+            )
```

Comment:
Please add payments list and payments cancel as well.

Reply:
the /list method is for #[cfg(feature = "olap")]
How do I conditionally add that service to that scope?

Reply:
You should be able to do something like this (refer to the surrounding code for more context):

https://github.com/juspay/hyperswitch/blob/22a5372481bbf854cffb8b683606cdf0644a5f54/crates/router/src/routes/app.rs#L86-L89